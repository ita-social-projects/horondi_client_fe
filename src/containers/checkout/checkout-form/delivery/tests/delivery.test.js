import React from 'react';
import Delivery from '../delivery';
import UkrpostAndCourier from '../ukrpost-and-courier';
import NovaPost from '../nova-post';
import SelfPickup from '../self-pickup';

let wrapper;
beforeEach(() => {
  wrapper = shallow(<Delivery deliveryType='NOVAPOSTCOURIER' />);
});

describe('Delivery component tests', () => {
  it('should render one <UkrpostAndCourier>', () => {
    expect(wrapper.find(UkrpostAndCourier)).toHaveLength(1);
  });
  it('should render zero <NovaPost>', () => {
    expect(wrapper.find(NovaPost)).toHaveLength(0);
  });
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should render one <SelfPickup />', () => {
    const wrapper = shallow(<Delivery deliveryType='SELFPICKUP' />);
    expect(wrapper.find(SelfPickup)).toHaveLength(1);
  });
});
