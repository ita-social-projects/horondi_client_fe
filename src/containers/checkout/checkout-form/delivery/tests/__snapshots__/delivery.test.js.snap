// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delivery component tests should match snapshot 1`] = `
<Fragment>
  <UkrpostAndCourier
    deliveryType="NOVAPOSTCOURIER"
  />
</Fragment>
`;
